llvm/include/llvm/TableGen/Record.h

```diff
@@ -2112,14 +2112,22 @@ struct LessRecordByID {
   }
 };
 
-/// Sorting predicate to sort record pointers by their
-/// name field.
+/// Sorting predicate to sort record pointers by their Name field.
 struct LessRecordFieldName {
   bool operator()(const Record *Rec1, const Record *Rec2) const {
     return Rec1->getValueAsString("Name") < Rec2->getValueAsString("Name");
   }
 };
 
+/// Sorting predicate to sort record pointers by their Name field, and break
+/// ties using record name.
+struct LessRecordFieldNameAndName {
```

Comment:
I kept it close to where other similar such comparators are defined, so they are all together.

Reply:
I agree with @tmatheson-arm. This comparator has nothing to do with TableGen core functionality and should be moved out of llvm/TableGen into llvm/util/TableGen where backends reside. Other comparators probably ended up here because there was not other common place to put them. Ideally, they (and especially LessRecordRegister) would be moved out, too.

Reply:
SG, I'll move it.